### PR TITLE
Extract some common logic out of VectorValue

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

###  What have you changed?

This PR is a preparation for https://github.com/tikv/tikv/pull/6132, extracting some common logic out to a better place, to reduce future code duplication.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Unit test
- Integration test

### CODE REVIEW

**Comment by breeswish** [comment]
@niedhui PTAL, thanks!

File: components/tikv_util/Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ prost-codec = ["prost"]
 [dependencies]
 backtrace = "0.3.9"
 byteorder = "1.2"
+codec = { path = "../codec" }
```
Comment (niedhui) [inline components/tikv_util/Cargo.toml]: Seems this PR does not change codes in  `tikv_util`, do we need this?

Comment (niedhui) [reply components/tikv_util/Cargo.toml]: Never mind. I saw we need this for BufferVec in #6323

**Pull request merged**